src/FileWriterTask.cxx

```diff
@@ -29,7 +32,14 @@ Source SourceFactory_by_FileWriterTask::create(string topic,
 
 std::vector<DemuxTopic> &FileWriterTask::demuxers() { return _demuxers; }
 
-FileWriterTask::FileWriterTask() { impl.reset(new FileWriterTask_impl); }
+FileWriterTask::FileWriterTask() {
+  using namespace std::chrono;
+  _id = duration_cast<nanoseconds>(system_clock::now().time_since_epoch())
```

Comment:
You are casting `long` to `uint64_t` here, it could be made explicit with a `static_cast`.